### PR TITLE
makefile improvements and headers cleanup

### PR DESCRIPTION
**Pull request by Lapshin**
This MR is related to our discussion https://sourceware.org/pipermail/binutils/2022-October/124060.html

I also prepared [your patch](https://sourceware.org/pipermail/binutils/2017-May/098159.html) rebased on the latest master branch.
Find it here - https://github.com/espressif/binutils-gdb/commits/feature/xtensa-dynamic-config. I have tested it and it works fine. You mentioned that you had some thoughts to implement... And you can also ask me to do that.

### CODE REVIEW

File: include/xtensa-dynconfig.h [hunk 1]
```diff
@@ -15,75 +15,80 @@
...
    Foundation, 51 Franklin Street - Fifth Floor, Boston, MA 02110-1301, USA.  */
 
-#ifndef XTENSA_DYNAMIC_CONFIG_H
-#define XTENSA_DYNAMIC_CONFIG_H
+#ifndef XTENSA_DYNCONFIG_H
+#define XTENSA_DYNCONFIG_H
+
+#include <stdarg.h>
+#include <stdio.h>
+#include "xtensa-config.h"
```
Comment (jcmvbkbc) [inline include/xtensa-dynconfig.h]: Why are these needed? There's nothing in this header that depends on any of these headers.

File: include/xtensa-tdep.h [hunk 2]
```diff
@@ -69,7 +71,7 @@ typedef enum
   xtRegisterGroupCP4	    = 0x10000000,    /* CP4.  */
   xtRegisterGroupCP5	    = 0x20000000,    /* CP5.  */
   xtRegisterGroupCP6	    = 0x40000000,    /* CP6.  */
-  xtRegisterGroupCP7	    = 0x80000000,    /* CP7.  */
+  xtRegisterGroupCP7	    = (int)0x80000000    /* CP7.  */
```
Comment (jcmvbkbc) [inline include/xtensa-tdep.h]: Why is this needed? This is different from what's in the binutils gdb/xtensa-tdep.h and this file is supposed to be copied to gdb, it's unused otherwise.

File: include/xtensa-dynconfig.h (hunk 1 above)
Comment (Lapshin) [reply include/xtensa-dynconfig.h]: Removed, thanks!

File: include/xtensa-tdep.h (hunk 2 above)
Comment (Lapshin) [reply include/xtensa-tdep.h]: Oops, it appeared accidentally. This file is synced now with binutils-gdb master branch.

**Comment by jcmvbkbc** [comment]
I've split the first patch in this series into several pieces and applied most of it.
I've applied the second patch of this series.
I've also applied the versioning scheme change and pushed the result here: https://github.com/jcmvbkbc/xtensa-dynconfig/commits/feature/makefile-improvements-and-code-cleanup

I've made corresponding changes to the binutils, took your gdb patch and fixed couple places there and pushed the result here: https://github.com/jcmvbkbc/binutils-gdb-xtensa/tree/xtensa-plugin-env

I've made corresponding changes to the gcc and pushed the result here: https://github.com/jcmvbkbc/gcc-xtensa/commits/xtensa-plugin-env

Please take a look/give it a try and let me know whether it looks good to you.
I'll then merge changes in this repo to the master and will begin upstreaming gcc and binutils changes.

**Comment by Lapshin** [comment]
> Please take a look/give it a try and let me know whether it looks good to you. I'll then merge changes in this repo to the master and will begin upstreaming gcc and binutils changes.

I tested a toolchain with your changes and it works well. The changes look good to me.

**Comment by jcmvbkbc** [comment]
Thanks a lot. Merged to master.
